### PR TITLE
[luci-interpreter] Add LogicalNot kernel

### PR DESCRIPTION
**Pull request by underflow101**
Fired Issue: #4764 

This commit adds `LogicalNot` kernel.

This commit have been tested on my PC,
and worked well by passing `./nncc build` & `./nncc test`.

Please review this PR, @seanshpark @jinevening @struss .
I'll be happy to get feedback and change to make improvement.

Thank you.

(Derived from SOS Mine Project)

Signed-off-by: underflow101 <ikarus125@gmail.com>

### CODE REVIEW

**Comment by underflow101** [comment]
@nnfw-bot test nncc-release

**Comment by underflow101** [comment]
@struss , @binarman ,
PTAL? :)
Thank you so much!

**Review by struss — APPROVED** [review]
LGTM

**Comment by seanshpark** [comment]
Code has `LogicalOr` but title is `LogicalNot `

**Review by binarman — APPROVED** [review]
LGTM

File: compiler/luci-interpreter/src/kernels/LogicalNot.test.cpp [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+  Tensor output_tensor = makeOutputTensor(DataType::S32);
+
+  LogicalNot kernel(&input_tensor, &output_tensor);
+  EXPECT_ANY_THROW(kernel.configure());
+}
+
+TEST(LogicalNotTest, InputTypeInvalid_NEG)
+{
+  Tensor input_tensor = makeInputTensor<DataType::S32>({1, 1, 1, 4}, {1, 0, 0, 1});
+  Tensor output_tensor = makeOutputTensor(DataType::BOOL);
```
[the remark this answers is not in the document]
Comment (binarman) [reply compiler/luci-interpreter/src/kernels/LogicalNot.test.cpp:57]: This is almost the same suggestion as in https://github.com/Samsung/ONE/pull/4767#pullrequestreview-519231016
It is not mandatory, I just think this sequence comply with `configure` + `execute` approach.

**Comment by underflow101** [comment]
> Code has `LogicalOr` but title is `LogicalNot `

I needed to resolve conflict, since `LogicalOr` was merged before-hand.
But still, this commit is about `LogicalNot` :)

File: compiler/luci-interpreter/src/kernels/LogicalNot.test.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (underflow101) [reply compiler/luci-interpreter/src/kernels/LogicalNot.test.cpp:57]: @binarman , Thank you for your advice :+1: 
I will test this as your feedback.

**Review by struss — APPROVED** [review]
LGTM

**Comment by seanshpark** [comment]
@underflow101 , OK to merge?

**Comment by underflow101** [comment]
> @underflow101 , OK to merge?

Yes, please :)

**Pull request merged**